hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/MutableVolumeSet.java

```diff
@@ -522,6 +539,19 @@ private void stopDiskChecker() {
     return ImmutableMap.copyOf(volumeStateMap);
   }
 
+  public boolean hasEnoughVolumes() {
+    int maxVolumeFailuresTolerated = conf.getInt(
+        DFSConfigKeysLegacy.DFS_DATANODE_FAILED_VOLUMES_TOLERATED_KEY,
+        DFSConfigKeysLegacy.DFS_DATANODE_FAILED_VOLUMES_TOLERATED_DEFAULT);
+
+    if (maxVolumeFailuresTolerated ==
+        DataNode.MAX_VOLUME_FAILURE_TOLERATED_LIMIT) {
+      return getVolumesList().size() >= 1;
+    } else {
+      return getFailedVolumesList().size() <= maxVolumeFailuresTolerated;
```

Comment:
What's the difference of if else here,   could you do a little explaination?

Reply:
This logic is borrowed from hdfs, when we set this 'tolerated' to -1, we mean unlimited number of bad volumes but we still should have at least 1 good volume left.
Otherwise, we should have fewer or equal number of bad volumes.

Reply:
@bshashikant sure, will cp the lines above into code comments.